sys/net/gnrc/netreg/gnrc_netreg.c

```diff
@@ -38,15 +39,18 @@ void gnrc_netreg_init(void)
 
 int gnrc_netreg_register(gnrc_nettype_t type, gnrc_netreg_entry_t *entry)
 {
-#if defined(MODULE_GNRC_NETAPI_MBOX) || defined(MODULE_GNRC_NETAPI_CALLBACKS)
 #ifdef DEVELHELP
-    /* only threads with a message queue are allowed to register at gnrc */
-    assert((entry->type != GNRC_NETREG_TYPE_DEFAULT) ||
-           sched_threads[entry->target.pid]->msg_array);
+    bool has_msg_q = sched_threads[entry->target.pid]->msg_array;
+
+#if defined(MODULE_GNRC_NETAPI_MBOX) || defined(MODULE_GNRC_NETAPI_CALLBACKS)
+    has_msg_q = has_msg_q || (entry->type != GNRC_NETREG_TYPE_DEFAULT);
 #endif
-#else
     /* only threads with a message queue are allowed to register at gnrc */
-    assert(sched_threads[entry->target.pid]->msg_array);
+    if (!has_msg_q) {
+        LOG_ERROR("\n!!!! gnrc_netreg: initialize message queue of thread %u "
```

Comment:
This is surrounded with `DEVELHELP`. And no. At the moment it fails an assertion. Which leads to newcomers and students reporting Kernel Panics on mailinglists and in our offices.